utils/dynstream/stream.go

```diff
@@ -146,12 +149,16 @@ func (s *stream[A, P, T, D, H]) receiver() {
 	for {
 		event, ok := buffer.FrontRef()
 		if !ok {
-			e, ok := <-s.inChan
-			if !ok {
+			select {
```

Comment:
Based on previous test results, converting channel send operations to `select` operations in this critical path that handles large volumes of events has shown significant performance impact (with substantial increase in CPU usage). Do we have any other viable solutions to address this problem?